### ISSUE TITLE
Switching colorschemes break after a few tries

### ISSUE THREAD

**Issue by andradei** [xolox/vim-colorscheme-switcher#6]
I've seen your explanation on how to circumvent Vim's limitation in this issue from a years ago https://github.com/altercation/solarized/issues/102

But after switching color schemes a few times with `F8`, the colors start to break. I either have to exit and renter Vim or close and reopen the buffer to get the colors showing properly.

This might be Vim's bug. But I appreciate any direction to try to fix this.

**Comment by justgage** [comment]
I've noticed this as well. I'm pretty sure it works like a big bunch of variables that are left set by previous color schemes thus causing the previous color scheme to pollute the next one. I'm not sure if there's an official way to clear them, we could possibly set up a "reset" color scheme that would set everything to a neutral color. However I'm not sure if it's possible. 

*disclaimer: I'm just speculating here I'm not sure how vim colorschemes work.*

**Comment by xolox** [comment]
Hi @andradei and @justgage and thanks for the feedback! As mentioned in the readme:

> Probably this solution is still not perfect, but it's a lot better than the behavior out of the box: Before I implemented the steps above, when I would cycle through my color schemes, Vim would eventually end up with black text on a white background and nothing else! With the steps above, I can cycle as many times as I want and **all of the color schemes I've checked so far look fine.**

As the bit I emphasized above states, it doesn't surprise me that my workaround is not complete (it's impossible for me to check compatibility with all of the Vim color schemes in existence :-).

One thing that @andradei mentioned is interesting though:

> But after switching color schemes a few times with F8, the colors start to break. I either have to exit and renter Vim or **close and reopen the buffer to get the colors showing properly.**

Indeed restarting Vim should fix things, but it surprises me that "closing and reopening a buffer" would fix anything. Can you specify exactly how you do this (which Vim commands you use) to make sure we're understanding each other correctly?

What would also be interesting for me to know is whether executing the following command restores the syntax highlighting to its proper state:

```vim
:let &syntax = &syntax
```

One thing that would help to avoid a lot of back and forth regarding this issue is if I can test with the color scheme(s) that cause breakage for you two. I see two options here:

1. If you know which color scheme(s) cause this issue then you can share those specific color scheme(s) with me (e.g. a link where I can download the color scheme).
2. If you don't know which color scheme(s) cause this issue then you can also share a ZIP / tar archive with all of your color schemes so I can try to reproduce the issue.

**Comment by andradei** [comment]
Thanks for the willingness to solve this. I have some more interesting information to give you:
1. I can't reproduce the problem on MacVim anymore, only on Terminal or iTerm2
2. I specifically delete the buffer with `:bd` or `:bdelete` and recreate a new one rather then reopening (since is was deleted, can't be reopened...)
3. When it happens, `:let &syntax = &syntax` solves the problem, thanks for the command.

The problem seems to be unrelated to _one_ colorscheme, but perhaps to colorschemes that have more than 255 colors?

I'm using MacVim for the time being, no issues there.

Some extra info:
1. I'm on OS X as you might have guessed
2. On Terminal, `SHIFT + F8` doesn't work. `F8` works fine. Both work on MacVim though.
3. Maybe these issues are related only to OS X Yosemite, since it had a bunch of systems changes.

**Comment by xolox** [comment]
I haven't been able to reproduce the issue described here. Although I don't normally use terminal Vim and the color scheme switcher plug-in together (I mostly use Vim on dozens of server systems where I don't make it a habit to install all of my Vim plug-ins :-) I have tried it quite a bit because of this issue, but it works fine for me, also with color schemes using 256 colors (e.g. [xoria256](http://www.vim.org/scripts/script.php?script_id=2140)). This may be related to the fact that I use Linux instead of Mac OS X.

However I did just release version 0.5.1 which automatically runs `:let &syntax = &syntax` after switching color schemes, but only in terminal Vim, because that's where the problem(s) seem to manifest.
